### PR TITLE
Make sure `limactl list --json` to print line-delimited JSON

### LINKED ISSUE #1280: `limactl list --json` now returns an array instead of newline delimited JSON
### Description

Originally reported in https://github.com/abiosoft/colima/issues/572

I don't think this was intentional, and should be fixed before the next release, as it breaks backwards compatibility.

And I think `limactl inspect` should be changed accordingly for consistency.

### PR DESCRIPTION
**Pull request by AkihiroSuda**
Fix #1280
A regression in 75594fa33bd73a2cc80a389993c7573b4bf3ec58 (#1249)

### CODE REVIEW

File: pkg/textutil/textutil.go [hunk 1]
```diff
@@ -17,3 +22,24 @@ func ExecuteTemplate(tmpl string, args interface{}) ([]byte, error) {
...
+// TemplateFuncMap is a text/template FuncMap.
+var TemplateFuncMap = template.FuncMap{
+	"json": func(v interface{}) string {
+		var b bytes.Buffer
+		enc := json.NewEncoder(&b)
+		enc.SetEscapeHTML(false)
+		if err := enc.Encode(v); err != nil {
+			panic(fmt.Errorf("failed to marshal as JSON: %+v: %w", v, err))
+		}
+		return strings.TrimSpace(b.String())
```
Comment (jandubois) [inline pkg/textutil/textutil.go:35]: Why is the `TrimSpace` needed?

Comment (AkihiroSuda) [reply pkg/textutil/textutil.go:35]: Because otherwise it produces "\n\n"-delimited JSON lines

Comment (jandubois) [reply pkg/textutil/textutil.go:35]: Wouldn't it be clearer to show the intent with this:

```suggestion
		return strings.TrimSuffix(b.String(), "\n")
```

**Review by jandubois — APPROVED** [review]
LGTM

File: pkg/textutil/textutil.go (hunk 1 above)
Comment (AkihiroSuda) [reply pkg/textutil/textutil.go]: done

**Review by jandubois — APPROVED** [review]
Thanks, LGTM

**Pull request merged**